general/development/policies/deprecation/index.md

```diff
@@ -64,7 +64,7 @@ debugging('foobar() is deprecated. Please use foobar::blah() instead.', DEBUG_DE
 ```php
 #[\core\attribute\deprecated('foobar::blahv()', since: '4.4', mdl: 'MDL-XXXXX')]
 public function foobar(): void {
-    \core\deprecation::emit_deprecation_if_present([self, __FUNCTION__]);
+    \core\deprecation::emit_deprecation_if_present([self::class, __FUNCTION__]);
```

Comment:
Alternatively, use `$this`

```suggestion
    \core\deprecation::emit_deprecation_if_present([$this, __FUNCTION__]);
```

i think it's good to show a variety of options. Whilst `self` is wrong, we can show both `self::class` and `$this`.

Reply:
I like the idea of using both! So, as you suggested, I've replaced this with `$this` <3